### ISSUE TITLE
issue creating and loading IR for EfficientNetB0 and MobileNetV2

### ISSUE THREAD

**Issue by gery9999** [openvinotoolkit/openvino#9896]
Hi!
I have been training an EfficientNetB0 (and a MobileNetV2 too). I have issues loading the model after conversion to OpenVINO IR.
My first guess was that I was using a wrong TF/Python version while training (TF2.6, Python version 3.9.).
However, also after training an EfficientNetB0-model in TF2.5, Python version 3.9, after I finally managed to convert the model to OpenVINO IR, I could not load the model using the model.xml and model.bin file.
Just for testing I converted and loaded a VGG16-model which was successful. But in the end I want to use EfficientNetB0 and MobileNetV2.
Is there a way to find out what is causing the issue? Probably it is a layer in EfficientNetB0/MobileNetV2 that is not yet supported by OpenVINO? How can I make my EfficientNetB0-model and my MobileNetV2-model .xml and .bin file load and run?

I am not really sure how to formulate my questions here in this forum. Bare with me, i have never done this before. But after spending several weeks in openvino documentation and not getting any further I decided to try getting help here.

**Comment by Iffa-Meah** [comment]
Hi @gery9999 ,

OpenVINO does officially support the model EfficientNetB0. You may refer [here](https://docs.openvino.ai/latest/omz_models_model_efficientnet_b0.html). If you downloaded the model using the OpenVINO Model Downloader, you should be able to get a model named efficientnet-b0. This model is one of the EfficientNet models designed to perform image classification and was pre-trained in TensorFlow*. All the EfficientNet models have been pre-trained on the ImageNet image database.

This model is based on this [GitHub Repo](https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet).
This repo also provided checkpoint files for training purposes which you can refer to and probably compare with your training method.

MobileNetV2 is also officially supported by OpenVINO. You may refer to this [documentation](https://docs.openvino.ai/latest/omz_models_model_mobilenet_v2.html).

Could you share these:

1. Steps & commands that you did to train the model
2. IR conversion commands
3. Which OpenVINO inferencing sample that you used to load the trained model and its command
4. Your trained model file (if possible to share)

**Comment by gery9999** [comment]
ad 1. In short here is the code I used for training. Quite usual I guess.
`import tensorflow as tf
from tensorflow.keras import layers
from tensorflow.keras.applications import MobileNetV2

img_width = img_height = 224
num_classes = 2
num_epochs = 100
batch_size = 32

# Model using MobilNetV2 architecture
mobv2 = tf.keras.applications.MobileNetV2(
    input_shape=(img_width, img_height, 3),
    alpha=0.35,
    include_top=False,
    weights='imagenet')
model = tf.keras.Sequential()
model.add(mobv2)
model.add(layers.GlobalAveragePooling2D())
model.add(layers.Dense(num_classes, activation='softmax'))

# images are accessed using keras ImageDataGenerator(...) and flow_from_dataframe(...) for both training and validation set
history = model.fit(
    train_ds,
    epochs=num_epochs,
    steps_per_epoch=train_num_samples/batch_size,
    validation_data=val_ds,
    validation_steps=val_num_samples/batch_size,
    callbacks=callbacks,
    verbose=1)

model.save('C:\movilenetv2.hdf5')`

ad. 2. Afterwards I did the conversion to OpenVINO IR. There is an internal script that is used in my team. I run it in Anaconda Prompt after activation of an environment that is set up for open vino. Basically, it freezes the hdf5 model (to pb format) and then coverts it to IR with:
`python.exe path_to\model_optimizer_2021_4_1\mo_tf.py --saved_model_dir={saved_model_path} --output_dir={saved_model_path} \
      --input_shape ({input_shape[0]},{input_shape[1]},{input_shape[2]},{input_shape[3]}) \
        --data_type={data_type} --reverse_input_channels`

When running this skript I get the following output:
`C:\Users\ella\Anaconda3\envs\tf2_keras_vino\lib\site-packages\keras\backend.py:450: UserWarning: `tf.keras.backend.set_learning_phase` is deprecated and will be removed after 2020-10-11. To update it, simply pass a True/False value to the `training` argument of the `__call__` method of your layer or model.
  warnings.warn('`tf.keras.backend.set_learning_phase` is deprecated and '
2022-01-26 14:19:37.692846: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-01-26 14:19:38.137078: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 6012 MB memory:  -> device: 0, name: NVIDIA GeForce RTX 2060 SUPER, pci bus id: 0000:01:00.0, compute capability: 7.5
2022-01-26 14:19:46.557879: W tensorflow/python/util/util.cc:368] Sets are not currently considered sequences, but this may change in the future, so consider avoiding using them.
WARNING:absl:Function `_wrapped_model` contains input name(s) mobilenetv2_0.35_224_input with unsupported characters which will be renamed to mobilenetv2_0_35_224_input in the SavedModel.
WARNING:absl:Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 52). These functions will not be directly callable after loading.
Model saved as .pb-file.
Model input shape (1, 224, 224, 3)
Model Optimizer arguments:
Common parameters:
        - Path to the Input Model:      None
        - Path for generated IR:        C:\WBCAbnormData\debugOpenVino
        - IR output name:       saved_model
        - Log level:    ERROR
        - Batch:        Not specified, inherited from the model
        - Input layers:         Not specified, inherited from the model
        - Output layers:        Not specified, inherited from the model
        - Input shapes:         (1,224,224,3)
        - Mean values:  Not specified
        - Scale values:         Not specified
        - Scale factor:         Not specified
        - Precision of IR:      FP32
        - Enable fusing:        True
        - Enable grouped convolutions fusing:   True
        - Move mean values to preprocess section:       None
        - Reverse input channels:       True
TensorFlow specific parameters:
        - Input model in text protobuf format:  False
        - Path to model dump for TensorBoard:   None
        - List of shared libraries with TensorFlow custom layers implementation:        None
        - Update the configuration file with input/output node names:   None
        - Use configuration file used to generate the model with Object Detection API:  None
        - Use the config file:  None
[ WARNING ] Could not find the Inference Engine Python API. At this moment, the Inference Engine dependency is not required, but will be required in future releases.
[ WARNING ] Consider building the Inference Engine Python API from sources or try to install OpenVINO (TM) Toolkit using "install_prerequisites.sh"
Model Optimizer version:        2021.4.1-3926-14e67d86634-releases/2021/4
[ WARNING ]
Detected not satisfied dependencies:
        tensorflow: installed: 2.9.0-dev20220125, required: ~= 2.4.1
        numpy: installed: 1.22.1, required: < 1.20

Please install required versions of components or use install_prerequisites script
C:\Workspaces\MachineLearning\OpenVinoTools\model_optimizer_2021_4_1\install_prerequisites\install_prerequisites_tf2.bat
Note that install_prerequisites scripts may install additional components.
[ WARNING ]  Using fallback to produce IR.
[ SUCCESS ] Generated IR version 10 model.
[ SUCCESS ] XML file: C:\WBCAbnormData\debugOpenVino\saved_model.xml
[ SUCCESS ] BIN file: C:\WBCAbnormData\debugOpenVino\saved_model.bin
[ SUCCESS ] Total execution time: 34.78 seconds.
It's been a while, check for a new version of Intel(R) Distribution of OpenVINO(TM) toolkit here https://software.intel.com/content/www/us/en/develop/tools/openvino-toolkit/download.html?cid=other&source=prod&campid=ww_2021_bu_IOTG_OpenVINO-2021-4-LTS&content=upg_all&medium=organic or on the GitHub*`

ad 3. What is OpenVINO inferencing sample? The OpenVINO model (xml and bin file) was loaded with Intel Build 2021.4.1 as far as I know - or better, I tried to load it, but it did not work. Is there another way of loading and testing the OpenVINO model?

ad 4. I can share the model and the IR I got as well as the environments I trained and converted in. I can not upload them in this post due to file format restrictions. But I will create a repo and upload it there.

Thank you for your help and patience.

**Comment by Iffa-Meah** [comment]
Hi,

your IR model works from my side.

Inferred using [OpenVINO Benchmark Python Tool]:(https://docs.openvino.ai/latest/openvino_inference_engine_tools_benchmark_tool_README.html)

![benchmark](https://user-images.githubusercontent.com/79643476/151497933-72fe8166-98d3-4785-a8f8-f100a2e9a419.png)


Inferred using [Hello Classification Python Sample](https://docs.openvino.ai/latest/openvino_inference_engine_ie_bridges_python_sample_hello_classification_README.html)

![classification](https://user-images.githubusercontent.com/79643476/151498132-2ce8c47c-7b69-430f-a8c7-2b8cd33cccbc.png)


You may use the [Official OpenVINO Inference Engine Sample](https://docs.openvino.ai/latest/openvino_docs_IE_DG_Samples_Overview.html) that is suitable to your use case and infer with your model

**Comment by gery9999** [comment]
Hi Iffa-Meah!
Thanks for checking!
For some reason I can suddenly run the net today. I do not know what kept the net from working the past weeks. Not sure what has changed but at least the issue seems resolved now :)
